seerep-hdf5/seerep-hdf5-pb/test/pb-write-load-test.cpp

```diff
@@ -0,0 +1,310 @@
+  // imgMsg.labels_bb = bB2DLabeled;
+
+  // help 1 line
+  return imgMsg;
+}
+
+class pbWriteLoadTest : public testing::Test
+{
+protected:
+  static std::shared_ptr<std::mutex> hdf5FileMutex;
+  static std::string hdf5FileName;
+  static std::shared_ptr<HighFive::File> hdf5File;
+  static std::shared_ptr<seerep_hdf5_pb::Hdf5PbImage> imageIO;
+
+  // static std::string projectUUID;
+  static boost::uuids::uuid projectUUID;
+  static boost::uuids::uuid messageUUID;
+  static std::string projectName;
+
+  static seerep::Image writeImage;
+  static seerep::Image readImage;
+
+  // Because the tests only compare the written and read data, we create the
+  // resources only once and share them between the tests
+  static void SetUpTestSuite()
+  {
+    projectName = "testProject";
+    projectUUID = boost::uuids::random_generator()();
+    messageUUID = boost::uuids::random_generator()();
+
+    hdf5FileMutex = std::make_shared<std::mutex>();
+    hdf5FileName = boost::lexical_cast<std::string>(projectUUID) + ".h5";
+    hdf5File = std::make_shared<HighFive::File>(hdf5FileName, HighFive::File::ReadWrite | HighFive::File::Create);
+
+    // temporary files are stored in the build directory of seerep-hdf5-pb
+    // auto seerepCore = std::make_shared<seerep_core::Core>("./", false);
+    // seerep_core_msgs::ProjectInfo projectInfo;
+    // projectInfo.name = projectName;
+    // projectInfo.uuid = projectUUID;
+    // seerepCore->createProject(projectInfo);
+
+    imageIO = std::make_shared<seerep_hdf5_pb::Hdf5PbImage>(hdf5File, hdf5FileMutex);
+    if (imageIO == nullptr)
+    {
+      GTEST_FATAL_FAILURE_("Error: Can't create HDF5Image object for writing images");
+    }
+
+    writeImage = createImageMessage(256, 256, boost::lexical_cast<std::string>(projectUUID),
+                                    boost::lexical_cast<std::string>(messageUUID));
+    // if (writeImage == NULL)
+    // {
+    //   GTEST_FATAL_FAILURE_("Error: No image data to write into HDF5 file");
+    // }
+
+    imageIO->writeImage(boost::lexical_cast<std::string>(messageUUID), writeImage);
+
+    auto gRPCImage = imageIO->readImage(boost::lexical_cast<std::string>(messageUUID));
+
+    if (!gRPCImage.has_value())
+    {
+      GTEST_FATAL_FAILURE_("Error: No data could be read from HDF5 file");
+    };
+
+    readImage = gRPCImage.value();
+  }
+
+  static void TearDownTestSuite()
+  {
+    std::filesystem::remove(hdf5FileName);
+  }
+};
+
+std::shared_ptr<std::mutex> pbWriteLoadTest::hdf5FileMutex;
+std::shared_ptr<HighFive::File> pbWriteLoadTest::hdf5File;
+std::string pbWriteLoadTest::hdf5FileName;
+
+boost::uuids::uuid pbWriteLoadTest::projectUUID;
+boost::uuids::uuid pbWriteLoadTest::messageUUID;
+std::string pbWriteLoadTest::projectName;
+
+std::shared_ptr<seerep_hdf5_pb::Hdf5PbImage> pbWriteLoadTest::imageIO;
+
+seerep::Image pbWriteLoadTest::writeImage;
+seerep::Image pbWriteLoadTest::readImage;
+
+TEST_F(pbWriteLoadTest, testImageHeader)
```

Comment:
Check if code docu can be applied to the test functions and write it if applicable